langchain-core/src/vectorstores.ts

```diff
@@ -58,12 +58,15 @@ export type VectorStoreRetrieverInput<V extends VectorStoreInterface> =
     );
 
 export interface VectorStoreRetrieverInterface<
-  V extends VectorStoreInterface = VectorStoreInterface
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  V extends VectorStoreInterface<Metadata> = VectorStoreInterface<any>,
```

Comment:
Not sure that's needed because `VectorStoreInterface<Metadata>` would already enforce that `any` actually needs to extend `Metadata`. Also, I would be hesitant to change the order of generics as others might be using this interface already, and their typings would break

Reply:
Oh, yeah you're right. Goofed and thought that was new. Seems this is the best way forward then.